pkg/types/chain_reader.go

```diff
@@ -53,12 +58,10 @@ type ChainReader interface {
 	// Contract call errors are returned in the Err field of BatchGetLatestValuesResult.
 	BatchGetLatestValues(ctx context.Context, request BatchGetLatestValuesRequest) (BatchGetLatestValuesResult, error)
 
-	// Bind will override current bindings for the same contract, if one has been set and will return an error if the
-	// contract is not known by the ChainReader, or if the Address is invalid
-	Bind(ctx context.Context, bindings []BoundContract) error
-
 	// QueryKey provides fetching chain agnostic events (Sequence) with general querying capability.
 	QueryKey(ctx context.Context, contractName string, filter query.KeyFilter, limitAndSort query.LimitAndSort, sequenceDataType any) ([]Sequence, error)
+	// ReplaySequence makes sure that sequence data that belongs to contractName/key pair is available for querying from blockID.
```

Comment:
given that this is not actually replaying, a better name would ~ `SetReadBlock` or `SetBlockCursor` or something that more explicit about what is happening

Add an example usage, too. IE is seems like the caller must call QueryKeys after this method for something useful to occur(?).

Reply:
It is "replaying" the underlying chain component so that it allows to `QueryKey` a sequence that happened before `Bind`

Reply:
Should be better now

Reply:
>It is "replaying" the underlying chain component so that it allows to QueryKey a sequence that happened before Bind

This is leaky. the caller isn't supposed to think or know about the underlying chain.

the name should mean something WRT to this api, not the chain. Setting the starting point for Reads is the scope for the caller.

Reply:
Maybe something like `EnableQueryingFrom()` ?